server/configuration-validation.js

```diff
@@ -182,13 +182,10 @@ function CollectionConfig({ collectionName, configuration }) {
 
   return {
     validateIndexName() {
-      // indexName is either undefined or a none empty string
-      if (
-        (indexName !== undefined && typeof indexName !== 'string') ||
-        indexName === ''
-      ) {
+      // indexName is either undefined or a non empty array of string
+      if (indexName !== undefined && !Array.isArray(indexName)) {
         log.error(
-          `The "indexName" option of "${collectionName}" should be a non-empty string`,
+          `The "indexName" option of "${collectionName}" should be a non-empty array of strings`,
```

Comment:
Could you write a few lines here as a comment to be added to the CHANGELOG, so people will be able to apply the change needed?